### PR TITLE
ARROW-8231: [Rust] Parse parquet key_value_metadata

### PR DESCRIPTION
**Pull request by jhorstmann**
The parquet-format FileMetaData struct contains optional key value pairs with additional metadata about the schema.

When the parquet file was generated using the java avro parquet writer, this for example contains the original avro schema under the `parquet.avro.schema` or `avro.schema` keys.

Please note that I'm still relatively new to rust, any feedback is appreciated.

### CODE REVIEW

File: rust/parquet/src/file/reader.rs [hunk 1]
```diff
@@ -261,6 +263,24 @@ impl<R: ParquetReader> SerializedFileReader<R> {
             None => None,
         }
     }
+
+    fn parse_key_value_metadata(key_value_metadata: Option<Vec<KeyValue>>) -> Option<HashMap<String, String>> {
+        match key_value_metadata {
+            Some (key_values) => {
+                let map : HashMap<String, String> = key_values.into_iter()
+                    .filter(|kv| kv.value.is_some()) // skip keys without corresponding value
+                    .map(|kv| (kv.key, kv.value.unwrap()))
```
Comment (vertexclique) [inline rust/parquet/src/file/reader.rs]: `filter_map` is doing exactly this. Let's use that.

File: rust/parquet/src/arrow/schema.rs [hunk 2]
```diff
@@ -61,12 +62,15 @@ where
         }
     }
 
+    let metadata = metadata.map(|m| m.clone())
+        .unwrap_or_else(|| HashMap::new());
```
Comment (vertexclique) [inline rust/parquet/src/arrow/schema.rs]:
```suggestion
        .unwrap_or(HashMap::default());
```

**Comment by vertexclique** [comment]
It would be nice to run the linter if they exist in here.

**Comment by jhorstmann** [comment]
Seems I wasn't on the latest master and parts of this are already implemented, will have to rebase.

**Comment by paddyhoran** [comment]
@jhorstmann running `cargo +stable fmt --all` should fix all the lint errors in CI

**Review by paddyhoran — APPROVED** [review]
LGTM pending CI

**Pull request closed without merging**